spring-cloud-kubernetes-commons/src/main/java/org/springframework/cloud/kubernetes/commons/config/ConfigMapConfigProperties.java

```diff
@@ -72,10 +77,16 @@ public void setSources(List<Source> sources) {
 	 */
 	public List<NormalizedSource> determineSources() {
 		if (this.sources.isEmpty()) {
-			return Collections.singletonList(new NormalizedSource(name, namespace));
+			if (useNameAsPrefix) {
+				LOG.warn(
+						"'spring.cloud.kubernetes.config.useNameAsPrefix' is set to 'true', but 'spring.cloud.kubernetes.config.sources'"
+								+ " is empty; as such will default 'useNameAsPrefix' to 'false'");
+			}
+			return Collections.singletonList(new NormalizedSource(name, namespace, ""));
```

Comment:
`return Collections.singletonList(new NormalizedSource(name, namespace, ""));`

means that users that do not opt-in this functionality will get the previous logic; so that nothing changes.